samples/ApptentiveSample/MainActivity.cs

```diff
@@ -5,45 +5,61 @@
 using System;
 using Android.Content;
 using Android.Gms.Common;
-using Firebase.Messaging;
-using Firebase.Iid;
 using Android.Util;
+using Kotlin;
+using Xamarin.Essentials;
+using Android.Support.V7.App;
+using AndroidX.Fragment.App;
+using AndroidX.Lifecycle;
+using Firebase.Messaging;
+using Firebase;
 
 namespace ApptentiveSample
 {
-    [Activity(Label = "ApptentiveSample", MainLauncher = true, Icon = "@mipmap/icon")]
-    public class MainActivity : Activity, IUnreadMessagesListener
+    [Activity(Label = "ApptentiveSample", MainLauncher = true)]
+    public class MainActivity : AppCompatActivity, IApptentiveActivityInfo
     {
         TextView unreadMessagesTextView;
 
+        
         protected override void OnCreate(Bundle savedInstanceState)
         {
             base.OnCreate(savedInstanceState);
-
+            Xamarin.Essentials.Platform.Init(this, savedInstanceState);
             // Set our view from the "main" layout resource
             SetContentView(Resource.Layout.Main);
+            
+            ApptentiveSDK.Android.Apptentive.RegisterApptentiveActivityInfoCallback(this);
 
             // Controls
             var eventNameEditText = FindViewById<EditText>(Resource.Id.eventEditText);
 
             var messageCenterButton = FindViewById<Button>(Resource.Id.messageCenterButton);
             messageCenterButton.Click += delegate
             {
-                Apptentive.ShowMessageCenter(this, (shown) => Console.WriteLine("Message Center shown: " + shown));
+                ApptentiveSDK.Android.Apptentive.ShowMessageCenter((shown) => Console.WriteLine("Message Center shown: " + shown));
             };
 
             var engageButton = FindViewById<Button>(Resource.Id.engageButton);
             engageButton.Click += delegate
             {
                 var eventName = eventNameEditText.Text;
-                Apptentive.Engage(this, eventName, (engaged) => Console.WriteLine("Interaction engaged: " + engaged));
+                Console.WriteLine("Event Name: " + eventName);
+                ApptentiveSDK.Android.Apptentive.Engage(eventName, null, (engaged) => Console.WriteLine("Interaction engaged: " + engaged));
             };
 
             var canShowInteractionButton = FindViewById<Button>(Resource.Id.canShowInteractionButton);
             canShowInteractionButton.Click += delegate
             {
                 var eventName = eventNameEditText.Text;
-                Apptentive.QueryCanShowInteraction(eventName, (canShowInteraction) => Toast.MakeText(this, canShowInteraction ? "Yes" : "No", ToastLength.Long).Show());
+                var canShowInteraction = ApptentiveSDK.Android.Apptentive.CanShowInteraction(eventName);
```

Comment:
Could you use wrapper methods here? As this sample app will be available along with the plugin repository, it also serves as a documentation. Though `ApptentiveSDK.Android.Apptentive.CanShowInteraction` would work I guess that is not something we would want them to do.